src/ensembl/src/content/app/species/state/general/speciesGeneralHelper.ts

```diff
@@ -660,38 +660,51 @@ export const getStatsForSection = (props: {
       })
     : undefined;
 
+  const isExpandedContentSame =
+    Object.keys(filteredData).length === summaryStatsKeys?.length;
+
+  if (isExpandedContentSame) {
+    return {
+      section,
+      summaryStats,
+      exampleLinks
+    } as StatsSection;
+  }
+
+  const processedGroups = groups.map((group) => {
+    const groupStats = groupsStatsMap[group];
+
+    const processedStats = groupStats
+      .map((subGroupStats) => {
+        const processedSubGroupStats: IndividualStat[] = [];
+
+        subGroupStats.forEach((stat) => {
+          if (filteredData[stat] || filteredData[stat] === 0) {
```

Comment:
I'm confused. Haven't we [guaranteed, by this point](https://github.com/Ensembl/ensembl-client/blob/fdfee6f80fe1bf5abe23f4a93d37edb999ff94b8/src/ensembl/src/content/app/species/state/general/speciesGeneralHelper.ts#L625-L633), that we are interested in everything that's in `filteredData`, so we do not need this extra if check?

Reply:
I also thought the same but not really. This is looping through subGroupStats which is not filtered.